test/grub2_test.rb

```diff
@@ -0,0 +1,159 @@
+require_relative "test_helper"
+
+require "bootloader/grub2"
+
+describe Bootloader::Grub2 do
+  before do
+    allow(::CFA::Grub2::Default).to receive(:new).and_return(double("GrubDefault").as_null_object)
+    allow(::CFA::Grub2::GrubCfg).to receive(:new).and_return(double("GrubCfg").as_null_object)
+    allow(Bootloader::Sections).to receive(:new).and_return(double("Sections").as_null_object)
+    allow(Yast::BootStorage).to receive(:available_swap_partitions).and_return([])
+    allow(Yast::BootStorage).to receive(:gpt_boot_disk?).and_return(false)
+  end
+
+  describe "#read" do
+    before do
+      allow(Yast::BootStorage).to receive(:device_map).and_return(double(empty?: false))
```

Comment:
I find quite confusing to find this here and then overridden in one of the only 2 `it` blocks that we have.